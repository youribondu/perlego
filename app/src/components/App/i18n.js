const translations = {
  fr: {
    title: "Livres qui pourraient te plaire",
  },
  en: {
    title: "Books You Might Like",
  },
};

export default translations;
