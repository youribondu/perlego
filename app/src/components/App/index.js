import React, { Component } from "react";
import Books from "../Books";
import "./index.scss";
import translations from "./i18n";
import LocaleContext from "../../services/context";
import { LOCALE_DEFAULT } from "../../settings/constants";

class App extends Component {
  render() {
    const { locale } = this.context ?? LOCALE_DEFAULT;

    return (
      <div className="p-app">
        <div className="p-app__books">
          <h1 data-testid="app-title">{translations[locale].title}</h1>
          <Books />
        </div>
      </div>
    );
  }
}

App.contextType = LocaleContext;

export default App;
