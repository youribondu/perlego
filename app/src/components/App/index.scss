@import "../../styles/index.scss";

.p-app {
  display: flex;
  justify-content: center;

  &__books {
    @include wrapper();
    width: 100%;
  }
}
