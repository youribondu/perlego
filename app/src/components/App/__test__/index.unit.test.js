import React from "react";
import ReactDom from "react-dom";
import App from "../index";
import { render, cleanup, act } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { create } from "react-test-renderer";

jest.mock("../../Books", () => () => <div></div>);

describe("App Component", () => {
  afterEach(cleanup);

  it("Renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<App />, div);
  });

  it("Renders app correctly", () => {
    let result = null;
    act(() => {
      result = render(<App />);
    });

    expect(result.getByTestId("app-title")).toHaveTextContent(
      "Books You Might Like"
    );
  });

  it("Matches snapshot empty", () => {
    const app = create(<App />).toJSON();
    expect(app).toMatchSnapshot();
  });
});
