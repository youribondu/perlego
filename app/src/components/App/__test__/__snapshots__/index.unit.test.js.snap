// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Matches snapshot empty 1`] = `
<div
  className="p-app"
>
  <div
    className="p-app__books"
  >
    <h1
      data-testid="app-title"
    >
      Books You Might Like
    </h1>
    <div />
  </div>
</div>
`;
